From fc16e27c7157d913225a5526bb33880dc53a5f49 Mon Sep 17 00:00:00 2001
From: Kuppuswamy Sathyanarayanan <sathyanarayanan.kuppuswamy@linux.intel.com>
Date: Tue, 22 May 2018 13:32:27 -0700
Subject: [PATCH 1025/1240] [REVERTME] usb: roles: intel_xhci: Set device mode
 as default.

In APL SOC, if there is no USB device connected to xHCI port, then
we are not able to override the mux mode in CFG0 register. So we
cannot switch the role after booting the system using sysfs file.
So forcing the default mode to USB_DEVICE, till we root cause this
iproblem.

Signed-off-by: Kuppuswamy Sathyanarayanan <sathyanarayanan.kuppuswamy@linux.intel.com>
---
 drivers/usb/roles/intel-xhci-usb-role-switch.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/usb/roles/intel-xhci-usb-role-switch.c b/drivers/usb/roles/intel-xhci-usb-role-switch.c
index d86d70e..dad2d19 100644
--- a/drivers/usb/roles/intel-xhci-usb-role-switch.c
+++ b/drivers/usb/roles/intel-xhci-usb-role-switch.c
@@ -167,6 +167,10 @@ static int intel_xhci_usb_probe(struct platform_device *pdev)
 	if (IS_ERR(data->role_sw))
 		return PTR_ERR(data->role_sw);
 
+	intel_xhci_usb_set_role(dev, USB_ROLE_HOST);
+	msleep(10);
+	intel_xhci_usb_set_role(dev, USB_ROLE_DEVICE);
+
 	return 0;
 }
 
-- 
2.7.4

